project/cmake/modules/FindFFMPEG.cmake

```diff
@@ -1,17 +1,164 @@
-include(ExternalProject)
-file(STRINGS ${CORE_SOURCE_DIR}/tools/depends/target/ffmpeg/FFMPEG-VERSION VER)
-string(REGEX MATCH "VERSION=[^ ]*$.*" FFMPEG_VER "${VER}")
-list(GET FFMPEG_VER 0 FFMPEG_VER)
-string(SUBSTRING "${FFMPEG_VER}" 8 -1 FFMPEG_VER)
-string(REGEX MATCH "BASE_URL=([^ ]*)" FFMPEG_BASE_URL "${VER}")
-list(GET FFMPEG_BASE_URL 0 FFMPEG_BASE_URL)
-string(SUBSTRING "${FFMPEG_BASE_URL}" 9 -1 FFMPEG_BASE_URL)
-
+if(FFMPEG_PATH)
+  set(ENABLE_INTERNAL_FFMPEG OFF)
+endif()
 
-if(ENABLE_INTERNAL_FFMPEG)
+# external FFMPEG
+if(NOT ENABLE_INTERNAL_FFMPEG)
   if(FFMPEG_PATH)
-    message(WARNING "Internal FFmpeg enabled, but FFMPEG_PATH given, ignoring")
+    set(ENV{PKG_CONFIG_PATH} "${FFMPEG_PATH}/lib/pkgconfig")
+    list(APPEND CMAKE_PREFIX_PATH ${FFMPEG_PATH})
+  endif()
+
+  # FFMPEG v3.1
+  set(REQUIRED_FFMPEG_VERSION 3.1)
+  set(FFMPEG_PKGS libavcodec>=57.48.101
```

Comment:
> are you suggesting that we should just accept any libs thrown at us?

that is not the case here because we use our fork as default.

> A dev that insists on building with a lower version can patch this.

I disagree. Development is a main use case that has to be considered. In general developers don't be experts on build system. hence this has to be kept as simple as possible.